lnst/Tests/Iperf.py

```diff
@@ -32,59 +32,47 @@ def run(self):
         server = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE,
                                   stderr=subprocess.PIPE, close_fds=True)
 
-        try:
-            stdout, stderr = server.communicate()
-            stdout = stdout.decode()
-            stderr = stderr.decode()
-        except KeyboardInterrupt:
-            pass
+        stdout, stderr = server.communicate()
+        stdout = stdout.decode().strip()
+        stderr = stderr.decode().strip()
+
+        if server.returncode > 0:
+            msg = f"iperf {self._role} returncode = {server.returncode}"
+            logging.error(msg)
+            logging.error(f"iperf stderr: {stderr}")
+            self._res_data["msg"] = msg
+            self._res_data["stderr"] = stderr
+            return False
 
         try:
             self._res_data["data"] = json.loads(stdout)
         except JSONDecodeError:
-            self._res_data["msg"] = "Error while parsing the iperf json output"
-            self._res_data["data"] = stdout
+            msg = "Error while parsing the iperf json output"
+            logging.error(msg)
+            logging.error(f"iperf stderr: {stderr}")
+            logging.debug(f"iperf stdout: {stdout}")
+            self._res_data["msg"] = msg
             self._res_data["stderr"] = stderr
-            logging.error(self._res_data["msg"])
             return False
 
-        try:
-            self._check_json_sanity()
-        except:
-            self._res_data["msg"] = "Iperf provided incomplete json data"
+        if not self._is_json_sane(self._res_data["data"]):
+            msg = "Iperf provided incomplete json data"
+            logging.error(msg)
+            logging.error(f"iperf stderr: {stderr}")
+            self._res_data["msg"] = msg
             self._res_data["data"] = stdout
```

Comment:
This was already in the code before this MR, but we shoud probably not overwrite something that is used for condition check.

When I checked the code further this statement is kind of redundant. The stdout must contain a valid json and is saved in `self._res_data["data"] = json.loads(stdout)` statement few lines above.

Reply:
Right, I removed the `self._res_data["data"] = stdout` line.